### ISSUE TITLE
CopyFile flattens output when current dir is same as input dir [0.11.3-beta]

### ISSUE THREAD

**Issue by martinvobr** [statiqdev/Statiq#245]
Hi,
it looks there are still some problems with refactored filesystem access when current dir is the same as the input dir.

Let's have following files:

```
/input/SomeDir/file-from-subdir.txt
/input/file-in-root.txt
/input/config.wyam
/input/run.cmd
/run.cmd
```

Config.wyam looks like this:

```
// Setup code

===
// Configuration code

---

Pipelines.Add("Resources",
    CopyFiles("*").Where(x => 
        Path.GetExtension(x) != ".cmd" 
        && Path.GetExtension(x) != ".wyam" 
        ));
```

Symptoms:
When I invoke run.cmd (`/run.cmd` file) output looks like this (which is correct):

```
output/file-in-root.txt
output/SomeDir/file-in-subdir.txt

```

Now change current dir to the 'input' and execute wyam by invoking run.cmd in the current dir (`/input/run.cmd` file). The output looks like this:

```
output/file-in-root.txt
output/file-in-subdir.txt
```

The directory structure is flatten.

Complete repro can be downloaded as usual from http://www.vobr.cz/wyam/wyam-repro-copy-files.zip.

**Comment by daveaglick** [comment]
Thanks for yet another great bug report. I'll get on this soon - trying hard to finish up the first phase of IO refactoring so I can stop chasing bugs based on the old code. Hopefully this will clear up on it's own, but I'll keep the issue open and run your repro with the new stuff just to be sure.

**Comment by martinvobr** [comment]
Always happy to help. If you want you can send me a note when next release is about to materialize and I can test it before it gets out. My usage scenario appears to be a bit different that yours so I can catch some bugs which would otherwise slip under the radar.

**Comment by martinvobr** [comment]
It looks fixed in 0.11.5-beta.

**Comment by daveaglick** [comment]
Verified this is still fixed in 0.12.0. Of note, changed the config file to the following (doesn't impact behavior, but required for breaking changes in 0.12.0):

```
Pipelines.Add("Resources",
    CopyFiles("**/*{!.cmd,!.wyam,}")
);
```
